### ISSUE TITLE
Add checks for incorrect use of OrdinalEncoder

### ISSUE THREAD

**Issue by ArturoAmorQ** [probabl-ai/skore#3029]
### Is your feature request related to a problem? Please describe.

If `OrdinalEncoder` is used, then `evaluate.checks` could either:
- tip on making sure categorical features are indeed ordered
- flag an "Issue" if the accuracy is close to the prevalence (cheap check) or permutation importance of ordinal-encoded features overlaps with zero (a bit more expensive as check).

```python
import pandas as pd
import skore
from sklearn.datasets import fetch_openml
from sklearn.compose import make_column_selector as selector
from sklearn.compose import make_column_transformer
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import OrdinalEncoder, StandardScaler
from sklearn.pipeline import make_pipeline

german_credit = fetch_openml(data_id=31, as_frame=True, parser="pandas")
X, y = german_credit.data, german_credit.target

numerical_columns_selector = selector(dtype_exclude="category")
categorical_columns_selector = selector(dtype_include="category")

numerical_columns = numerical_columns_selector(X)
categorical_columns = categorical_columns_selector(X)

categorical_preprocessor = OrdinalEncoder(handle_unknown="use_encoded_value",unknown_value=-1)
numerical_preprocessor = StandardScaler()

preprocessor = make_column_transformer(
    (categorical_preprocessor, categorical_columns),
    (numerical_preprocessor, numerical_columns),
)

model = make_pipeline(preprocessor, LogisticRegression(max_iter=500))
report = skore.evaluate(model, X, y, splitter=0.2)
report.checks.summarize()
```

 

### Describe the solution you'd like

I'd prefer flagging an "Issue" with a check on the permutation importance of ordinal-encoded features. That is more robust to pipelines with informative numerical features.

**Comment by GaetandeCast** [comment]
Good idea, thanks! I'll look into it.
